frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/active-health-checks.spec.ts

```diff
@@ -0,0 +1,63 @@
+import { checkErrors } from '../../../integration-tests-cypress/support';
+import { MINUTE } from '../consts';
+
+describe('Test Popover behaviour for different active health check cases.', () => {
+  before(() => {
+    cy.login();
+    cy.visit('/');
+    cy.install();
+    cy.visit('/ocs-dashboards/block-file');
+  });
+
+  after(() => {
+    checkErrors();
+    cy.logout();
+  });
+
+  const scaleDownAndOpenHC = (state: string, deployments: string[]) => {
+    const scaleTo = (replicas: number) =>
+      deployments.forEach((deployment) => {
+        cy.exec(`oc scale --replicas=${replicas} deploy ${deployment} -n openshift-storage`);
+      });
+    scaleTo(0);
+    cy.byTestID(`${state}-secondary-status`, { timeout: 2 * MINUTE }).should('be.visible');
+    cy.byTestID('Storage Cluster-health-item')
+      .contains('Storage Cluster')
+      .click();
+    return () => scaleTo(1);
+  };
+
+  const verifyMessages = (expectedMessages: string[]) => {
+    cy.byTestID('healthcheck-message', { timeout: 2 * MINUTE })
+      .should('have.lengthOf', expectedMessages.length)
+      .each((row, index, collection) => {
+        const expectedMessage = expectedMessages[collection.length - index - 1];
+        const actualMessage = row.text();
+        expect(actualMessage).toEqual(expectedMessage);
+      });
+  };
+
+  it('Popover shows all warnings.', () => {
+    const restoreResources = scaleDownAndOpenHC('Warning', ['rook-ceph-mon-a', 'rook-ceph-mgr-a']);
+    verifyMessages(['1/3 mons down, quorum b,c', 'no active mgr']);
+    restoreResources();
+  });
+
+  it('Popover shows the error.', () => {
+    const restoreResources = scaleDownAndOpenHC('Error', ['rook-ceph-mon-a', 'rook-ceph-mon-b']);
+    verifyMessages(['failed to get status. . timed out: exit status 1']);
+    restoreResources();
+  });
+
+  it('No Popover when cluster is healthy.', () => {
+    // Check if cluster is in a healthy state.
+    cy.byTestID('Warning-secondary-status', { timeout: 2 * MINUTE }).should('not.exist');
+    cy.byTestID('Error-secondary-status', { timeout: 2 * MINUTE }).should('not.exist');
```

Comment:
```suggestion
    cy.byTestID('storage-cluster-health-item-error-secondary-status', { timeout: 2 * MINUTE }).should('not.exist');
```

Reply:
by the way why is it secondary-status ? is there a primary status too ?

Reply:
I named it after the [component](https://github.com/rexagod/console/blob/a462dbee4868dbfb5e94a92edcba0dad7433cf8d/frontend/packages/console-shared/src/components/status/SecondaryStatus.tsx#L9) where it was defined. This is basically the status that shows below the "Storage Cluster" title on the OCS Dashboard.